scapy/tools/UTscapy/__init__.py

```diff
@@ -820,28 +799,34 @@ def main(argv):
         if VERB > 2:
             print("### Loading:", TESTFILE, file=sys.stderr)
         PREEXEC = PREEXEC_DICT[TESTFILE] if TESTFILE in PREEXEC_DICT else GLOB_PREEXEC
-        output, result, campaign = execute_campaign(open(TESTFILE), OUTPUTFILE,
-                                          PREEXEC, NUM, KW_OK, KW_KO,
-                                          DUMP, FORMAT, VERB, ONLYFAILED,
-                                          CRC, autorun_func, pos_begin, ignore_globals)
+        with open(TESTFILE) as testfile:
+            output, result, campaign = execute_campaign(testfile, OUTPUTFILE,
+                                                        PREEXEC, NUM, KW_OK, KW_KO,
+                                                        DUMP, FORMAT, VERB, ONLYFAILED,
+                                                        CRC, autorun_func, pos_begin,
+                                                        ignore_globals)
         runned_campaigns.append(campaign)
         pos_begin = campaign.end_pos
         if UNIQUE:
             glob_title = campaign.title
         glob_output += output
-        if not result:
+        if not result and BREAKFAILED:
             glob_result = 1
             break
 
     if VERB > 2:
             print("### Writing output...", file=sys.stderr)
     # Concenate outputs
     if FORMAT == Format.HTML:
-        glob_output = pack_html_campaigns(runned_campaigns, glob_output, LOCAL, glob_title)
-    
-    OUTPUTFILE.write(glob_output.encode("utf8", "ignore")
-                     if 'b' in OUTPUTFILE.mode else glob_output)
-    OUTPUTFILE.close()
+        glob_output = pack_html_campaigns(runned_campaigns, glob_output, glob_title)
+
+    if OUTPUTFILE == sys.stdout:
+        OUTPUTFILE.write(glob_output.encode("utf8", "ignore")
+                         if 'b' in OUTPUTFILE.mode else glob_output)
+    else:
+        with open(OUTPUTFILE, "wb") as f:
```

Comment:
In the case where this would fail (e.g. access denied), then scapy would fail after having performed all the tests: it would be good to check before performing all the tests that we can write in a such file

Reply:
Oh I see! I'm adding a check using `os.access`
Thanks!